### ISSUE TITLE
Different results on nine-species benchmark with your publications

### ISSUE THREAD

**Issue by qiyueliuhuo23** [ProteomeTeam/NovoB#5]
Hi, thanks for your amazing work on de novo peptide sequencing. I have been trying to reproduce your work and results on the nine-species benchmark with NovoB. I download the data and model weights from zenodo, but I got a much lower peptide accuracy on some datasets. For example, on mouse dataset, I got about 39%, but result in your paper is about 45%. I am trying to reproduce other results of the rest dataset. Can you help me to find out what makes the result lower?
Something might be helpful:
I am using RTX 2080ti with tensorflow 2.9 and keras 2.9.

**Comment by pardess** [comment]
We think you need to add some processing for PTMs in the peptide sequence.
When comparing peptide sequences, PTMs("C(+57.02)", "M(+15.99)", "N(+.98)", and "Q(+.98)") are replaced by "C", "m", "n", and "q", respectively.

**Comment by qiyueliuhuo23** [comment]
Because in Prediction.py, you did not write true peptides into result.txt, I wrote some codes to get true peptides, but I can not get higher peptide accuracy, even lower, close to zero.
Here is my code to get the ground truth.
``` python
def get_pep(mgf_path):
    peps = []
    peps_r = []
    mgf_f = mgf.read(mgf_path)
    for spec in mgf_f:
        peptide = spec['params']['seq']
        peptide = peptide.replace("C(+57.02)", "C").replace("M(+15.99)", "m").replace("N(+.98)", "n").replace("Q(+.98)", "q")
        peps.append(peptide)
        peps_r.append(peptide[::-1])
    return peps, peps_r
```
Here is the code to read predictions from result.txt
```python
predictions = []
predictions_r = []
with open("result.txt", 'r') as f:
    data = f.readline()
    while data:
        data = data.split('\'')
        predict = data[1]
        truth = data[3]
        predictions.append(predict)
        predictions_r.append(truth)
        data = f.readline()
```
Can you help me to figure out what is wrong with the code?

**Comment by pardess** [comment]
I am very sorry for not having a detailed explanation to preprocess spectra.
The spectrum needs to be filtered to match the index.
Spectra with masses greater than 3000 and spectra with more than 1000 peaks are excluded.

Please refer to the code below.

```python
Proton = 1.007276035
H2O = 18.0105647

def get_pep(mgf_path):
    peps = []
    peps_r = []
    mgf_f = mgf.read(mgf_path)
    for spec in mgf_f:
        charge = int(spec['params']['charge'][0])
        pmz_ = spec['params']['pepmass'][0]
        pmass = (pmz_ - Proton)*charge - H2O;
        mz = spec['m/z array']
        if pmass > 3000 or len(mz) > 1000 :
             continue
        peptide = spec['params']['seq']
        peptide = peptide.replace("C(+57.02)", "C").replace("M(+15.99)", "m").replace("N(+.98)", "n").replace("Q(+.98)", "q")
        peps.append(peptide)
        peps_r.append(peptide[::-1])
    return peps, peps_r
```

**Comment by qiyueliuhuo23** [comment]
Thanks your reply, the results are close to those in your publication now. Have a good day!
